src/traces/pie/attributes.js

```diff
@@ -79,7 +79,27 @@ module.exports = {
 
     text: {
         valType: 'data_array',
```

Comment:
That was on purpose - though probably never explicitly discussed. My rationale is that `text` is essentially the `x` data for pies, and they would be meaningless without this being an array of data. Can you think of a counterexample?